README.md

```diff
@@ -1,2 +1,307 @@
+
+```
+> bx cr namespace-add guestbook
+```
+
+This will create a registry url of `registry.ng.bluemix.net/guestbook`
+that you can use for uploading images.
+
+## Building using bx client
+
+The bx client can be used to build images using the local docker
+system. When building with the bx client all the authentication to the
+IBM Cloud Container Registry is handled behind the scenes.
+
+We will want to provide a meaningful name for the image, as well as
+some kind of version.
+
+```
+> bx cr build --tag registry.ng.bluemix.net/guestbook/gbv1:2018-05-18-1 guestbook
+Sending build context to Docker daemon  16.38kB
+Step 1/13 : FROM golang as builder
+ ---> 6b369f7eed80
+Step 2/13 : RUN go get github.com/codegangsta/negroni
+ ---> Using cache
+ ---> 6235665371cf
+Step 3/13 : RUN go get github.com/gorilla/mux github.com/xyproto/simpleredis
+ ---> Using cache
+ ---> c7b784b25fd7
+Step 4/13 : COPY main.go .
+ ---> 22bb5236a85b
+Step 5/13 : RUN go build main.go
+ ---> Running in 7809ad57b6b4
+ ---> 7b70b540d7d4
+Removing intermediate container 7809ad57b6b4
+Step 6/13 : FROM busybox:ubuntu-14.04
+ ---> d16744963217
+Step 7/13 : COPY --from=builder /go//main /app/guestbook
+ ---> 8b7eb0124777
+Step 8/13 : ADD public/index.html /app/public/index.html
+ ---> 9d763716676c
+Step 9/13 : ADD public/script.js /app/public/script.js
+ ---> 353bca474993
+Step 10/13 : ADD public/style.css /app/public/style.css
+ ---> 81819bed4684
+Step 11/13 : WORKDIR /app
+ ---> 1a99524a116a
+Removing intermediate container 020c16388591
+Step 12/13 : CMD ./guestbook
+ ---> Running in bdb88f981da7
+ ---> fc76ec26b945
+Removing intermediate container bdb88f981da7
+Step 13/13 : EXPOSE 3000
+ ---> Running in b04f65bdedad
+ ---> b63df6f0716e
+Removing intermediate container b04f65bdedad
+Successfully built b63df6f0716e
+Successfully tagged registry.ng.bluemix.net/guestbook/gbv1:2018-05-18-1
+The push refers to a repository [registry.ng.bluemix.net/guestbook/gbv1]
+cfd8baa1be3c: Pushed
+3d8c58449645: Pushed
+98a83cd4e753: Pushed
+9b5fbbfcbb1d: Pushed
+5f70bf18a086: Layer already exists
+5dbcf0efe4f2: Layer already exists
+4: digest: sha256:cca8cdb32ed616ae13087a9b557696896505bb36241c0f24588987094a687000 size: 1772
+
+OK
+```
+
+This builds and pushes all the relevant layers.
+
+The `--tag` field gives the image a name and version, separated by the
+`:`. When using the IBM Cloud Container Registry the name is going to
+include the full URL to that image.
+
+The version portion can be anything that's meaningful to you. The
+important thing is that tags do not get reused. If you rebuild an
+image, make sure to change the version before uploading. Many
+kubernetes primitives, such as deployments, will perform a graceful
+rolling upgrade when you specify a new image version. If image
+versions get reused it hampers the effectiveness of container
+orchestration systems.
+
+# Retrieving from IBM Cloud
+
+IBM Cloud images are included in kubernetes configuration in the same
```

Comment:
having trouble parsing this... my "included" did you mean "referenced" ?

Reply:
ack, will try to clear up